math/mathmore/src/GSLRndmEngines.cxx

```diff
@@ -83,7 +83,21 @@ gsl_ran_multivariate_gaussian(const gsl_rng *r, const gsl_vector *mu, const gsl_
 namespace ROOT {
 namespace Math {
 
-
+   // workspace used for caching when generating random distributions
+   // matrix and vector using in GaussianND
+   // in principle we can create several WS depending on the needs
+   struct GSLRndmInternalWS {
```

Comment:
I don't see this class used anywhere, is it really necessary? Maybe you used it in some intermediate development step, but in the end it's not needed?

Reply:
No it is not needed. I was thinking it to use to manage the gel_vector and matrices used in the code, but then I realised that you can create them as views (see line 271 of [here](https://github.com/root-project/root/pull/12615/files/01acfe1e55cefa654cbe4f5ead55d380ce562edb#diff-1ce96413def0c1fc349948047f7c5d8fa6ece7a0437e88f560187a4cb30edff7) ), so it is not needed. I will remove it